### PR TITLE
Questionsetitem form created

### PR DESCRIPTION
**Pull request by Midhun-h4r**
#30 
Question Set item form created.

### CODE REVIEW

File: resources/views/wisehabit/question_set_item.blade.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+@section('body')
+<div class="col-md-6 col-md-offset-3"><br>
+    <h2 class="text-center">Question Set Item Form</h2><br>
+    <div>@include('wisehabit.includes.messagebox')</div>
+    <form method="POST" action="{{route('createQuestionSetItem')}}">
+        @csrf
+        <div class="form-group col-xs-12">
+            <label for="question_set_id">Question Set</label>
+            <select class="form-control" name="question_set_id" id="question_set_id">
+                <option disabled @if(empty(old('question_set_id' ))) selected @endif>Choose Question Set Id</option>
```
Comment (rahulreghunath) [inline resources/views/wisehabit/question_set_item.blade.php:15]: Choose Question Set Id to Choose Question Set

File: resources/views/wisehabit/question_set_item.blade.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                @foreach($questionSets as $questionSet)
+                <option value="{{$questionSet->id}}" @if(old('question_set_id' )==$questionSet->id ) selected @endif>{{$questionSet->id}}</option>
+                @endforeach
+            </select>
+            <small class="text-danger">{{$errors->first('question_set_id')}}</small>
+        </div>
+        <div class="form-group col-xs-12">
+            <label for="question_id">Question </label>
+            <select class="form-control" name="question_id" id="question_id">
+                <option disabled @if(empty(old('question_id' ))) selected @endif>Choose Question Id</option>
```
Comment (rahulreghunath) [inline resources/views/wisehabit/question_set_item.blade.php:25]: Choose Question Id to Choose Question

File: app/Http/Requests/Wisehabit/QuestionSetItemRequest.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'question_set_id' => 'numeric',
```
Comment (rahulreghunath) [inline app/Http/Requests/Wisehabit/QuestionSetItemRequest.php:27]: question_set_id is not an optional field, so add required validation

**Review by rahulreghunath — CHANGES_REQUESTED** [review]
Edit function is not in the issue but partially done, good. Some small changes need to be done, otherwise, look fine.

**Comment by rahulreghunath** [comment]
Complete the edit function in the same pull request :smile:

File: routes/web.php [hunk 4]
```diff
@@ -95,4 +95,19 @@
...
+
+    Route::get('question-set-item/new', [
+        'uses' => 'QuestionSetItemController@index',
+        'as' => 'newQuestionSetItem'
+    ]);
+    Route::post('question-set-item/new', [
+        'uses' => 'QuestionSetItemController@createQuestionSetItem',
+        'as' => 'createQuestionSetItem'
+    ]);
+    Route::post('question-set-item/new/{id}', [
```
Comment (rahulreghunath) [inline routes/web.php:109]: url for editing should be `question-set-item/{id}/edit`

**Review by rahulreghunath — CHANGES_REQUESTED** [review]

File: resources/views/wisehabit/question_set_item.blade.php (hunk 2 above)
Comment (Midhun-h4r) [reply resources/views/wisehabit/question_set_item.blade.php:25]: done

File: app/Http/Requests/Wisehabit/QuestionSetItemRequest.php (hunk 3 above)
Comment (Midhun-h4r) [reply app/Http/Requests/Wisehabit/QuestionSetItemRequest.php]: Done

**Comment by rahulreghunath** [comment]
Edit for form and routes not added please add those asap.

**Comment by Midhun-h4r** [comment]
#37 
Edit for question set item form and routes  done.

**Comment by rahulreghunath** [comment]
No further change found. Might be files missing from the commit.

File: resources/views/wisehabit/edit_question_set_item.blade.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    <h2 class="text-center">Edit Question Set Item</h2><br>
+    <div>@include('wisehabit.includes.messagebox')</div>
+    <form method="POST" action="{{route('updateQuestionSetItem',['id'=>request()->segment(3)])}}">
+        @csrf
+        <div class="form-group col-xs-12">
+            <label for="question_id">Question Set</label>
+            <select class="form-control" name="question-set_id" id="question_set_id">
+                <option disabled>Choose Question Set</option>
+                @foreach($questions as $question)
+                <option value="{{$questionSet->id}}" @if(old('question_set_id',$oldData->question_set_id )==$questionSet->id ) selected @endif>{{$questionSet->title}}</option>
```
Comment (rahulreghunath) [inline resources/views/wisehabit/edit_question_set_item.blade.php:17]: Undefined variable: questionSet

did you tested it locally? :unamused:

File: resources/views/wisehabit/edit_question_set_item.blade.php [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+<div class="col-md-6 col-md-offset-3"><br>
+    <h2 class="text-center">Edit Question Set Item</h2><br>
+    <div>@include('wisehabit.includes.messagebox')</div>
+    <form method="POST" action="{{route('updateQuestionSetItem',['id'=>request()->segment(3)])}}">
+        @csrf
+        <div class="form-group col-xs-12">
+            <label for="question_id">Question Set</label>
+            <select class="form-control" name="question-set_id" id="question_set_id">
+                <option disabled>Choose Question Set</option>
+                @foreach($questions as $question)
```
Comment (rahulreghunath) [inline resources/views/wisehabit/edit_question_set_item.blade.php:16]: Loops questionSet instead of questions

**Review by rahulreghunath — CHANGES_REQUESTED** [review]

File: resources/views/wisehabit/edit_question_set_item.blade.php (hunk 6 above)
Comment (Midhun-h4r) [reply resources/views/wisehabit/edit_question_set_item.blade.php]: changed

**Comment by rahulreghunath** [comment]
**Route [editQuestionSetItem] not defined.**
Test locally and update

**Review by rahulreghunath — CHANGES_REQUESTED** [review]
Please test locally and update

**Comment by rahulreghunath** [comment]
Filed should not be edited directly in github

**Comment by rahulreghunath** [comment]
Also change form to new template

File: resources/views/wisehabit/edit_question_set_item.blade.php [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+@endsection
+@section('body')
+<div class="col-md-6 col-md-offset-3"><br>
+    <h2 class="text-center">Edit Question Set Item</h2><br>
+    <div>@include('wisehabit.includes.messagebox')</div>
+    <form method="POST" action="{{route('updateQuestionSetItem',['id'=>request()->segment(3)])}}">
+        @csrf
+        <div class="form-group col-xs-12">
+            <label for="question_id">Question Set</label>
+            <select class="form-control" name="question-set_id" id="question_set_id">
```
Comment (rahulreghunath) [inline resources/views/wisehabit/edit_question_set_item.blade.php:14]: incorrect field name `question-set_id`

**Review by rahulreghunath — CHANGES_REQUESTED** [review]

**Review by rahulreghunath — APPROVED** [review]

**Pull request merged**